spec/javascripts/modules/all-content-finder.spec.js

```diff
@@ -0,0 +1,151 @@
+        this.update = updateSpy
+      })
+
+      allContentFinder.init()
+
+      $taxonomySelect = fixture.querySelector('.js-all-content-finder-taxonomy-select')
+    })
+
+    it('initialises the taxonomy select on init and calls update', () => {
+      expect(GOVUK.TaxonomySelect).toHaveBeenCalledWith({ $el: $taxonomySelect })
+      expect(updateSpy).toHaveBeenCalledTimes(1)
+    })
+
+    it('calls update on the taxonomy select when a change occurs', () => {
+      const event = new Event('change')
+      $taxonomySelect.dispatchEvent(event)
+
+      expect(updateSpy).toHaveBeenCalledTimes(2) // Twice including the initial call on init()
+    })
+  })
+
+  describe('analytics tracking', () => {
+    beforeEach(() => {
+      spyOn(GOVUK.analyticsGa4.Ga4FinderTracker, 'trackChangeEvent')
+      spyOn(GOVUK.analyticsGa4.Ga4EcommerceTracker, 'init')
+    })
+
+    describe('when usage tracking is declined', () => {
+      beforeEach(() => {
+        GOVUK.setConsentCookie({ usage: false })
+        allContentFinder.init()
+      })
+
+      it('does not fire analytics tracking on form element changes', () => {
+        const event = new Event('change', { bubbles: true })
+        fixture.querySelector('#foo').dispatchEvent(event)
+        expect(GOVUK.analyticsGa4.Ga4FinderTracker.trackChangeEvent).not.toHaveBeenCalled()
+      })
+
+      it('does not set up ecommerce tracking', () => {
+        expect(GOVUK.analyticsGa4.Ga4EcommerceTracker.init).not.toHaveBeenCalled()
+      })
+    })
+
+    describe('when usage tracking is accepted', () => {
+      beforeEach(() => {
+        GOVUK.setConsentCookie({ usage: true })
+        allContentFinder.init()
+      })
+
+      it('fires analytics tracking on form element changes through the GA4 finder tracker', () => {
+        const event = new Event('change', { bubbles: true })
+        const input = fixture.querySelector('#foo')
+        input.dispatchEvent(event)
+
+        expect(GOVUK.analyticsGa4.Ga4FinderTracker.trackChangeEvent).toHaveBeenCalledWith(input, 'FooCategory')
+      })
+
+      it('sets up ecommerce tracking', () => {
+        expect(GOVUK.analyticsGa4.Ga4EcommerceTracker.init).toHaveBeenCalled()
+      })
+    })
+
+    describe('when usage tracking is accepted after component has already initialised', () => {
+      beforeEach(() => {
+        GOVUK.setConsentCookie({ usage: false })
+        allContentFinder.init()
+      })
+
+      it('fires analytics tracking on form element changes through the GA4 finder tracker', (done) => {
+        window.dispatchEvent(new Event('cookie-consent'))
+
+        setTimeout(() => {
+          const event = new Event('change', { bubbles: true })
+          const input = fixture.querySelector('#foo')
+          input.dispatchEvent(event)
+
+          expect(GOVUK.analyticsGa4.Ga4FinderTracker.trackChangeEvent).toHaveBeenCalledWith(input, 'FooCategory')
+          done()
+        }, 1) // allow for async event to propagate
```

Comment:
Are you sure you need this? I was under the impression dispatchEvent was synchronous and not sure there's any async code (same for below)

Reply:
Very good question. I had the tests crash and thought this fixed it, but maybe it was ~random solar flares~ a transient local issue because it works today 🤷‍♂️